src/java.base/share/classes/sun/security/pkcs/PKCS8Key.java

```diff
@@ -91,47 +95,78 @@ protected PKCS8Key() { }
      *
      * This method is also used by {@link #parseKey} to create a raw key.
      */
-    protected PKCS8Key(byte[] input) throws InvalidKeyException {
+    public PKCS8Key(byte[] input) throws InvalidKeyException {
         try {
             decode(new DerValue(input));
         } catch (IOException e) {
             throw new InvalidKeyException("Unable to decode key", e);
         }
     }
 
+    private PKCS8Key(byte[] privEncoding, byte[] pubEncoding)
+        throws InvalidKeyException {
+        this(privEncoding);
+        pubKeyEncoded = pubEncoding;
```

Comment:
If it isn't used anywhere, then it's probably from an old idea that I didn't completely clean up

Reply:
This constructor is eventually used by PEMEncoder for KeyPair.   Yes if the private key encoding did have a public key already it would be overwritten; however, the API usage is to combine the given KeyPair into PEM text.  The expected usage is after generating a KeyPair.
As far as preventing an overwrite, while it can probably be done, I don't see the value in adding a check at this point.

Reply:
I see. Good.